testsuites/syncgateway/upgrade/testserver/test_upgrade.py

```diff
@@ -0,0 +1,424 @@
+import os
+import random
+import time
+
+from concurrent.futures import ProcessPoolExecutor
+from couchbase.bucket import Bucket
+
+from keywords.couchbaseserver import verify_server_version
+from keywords.utils import log_info, host_for_url
+from keywords.SyncGateway import (verify_sync_gateway_version,
+                                  verify_sync_gateway_product_info,
+                                  SyncGateway)
+from keywords.ClusterKeywords import ClusterKeywords
+from keywords.MobileRestClient import MobileRestClient
+from keywords.constants import SDK_TIMEOUT
+from utilities.cluster_config_utils import persist_cluster_config_environment_prop
+
+from libraries.testkit.cluster import Cluster
+
+from CBLClient.Authenticator import Authenticator
+from CBLClient.Document import Document
+from CBLClient.Replication import Replication
+
+
+def test_upgrade(params_from_base_test_setup):
+    """
+    @summary
```

Comment:
Can you write full summary for this upgrade test, it will be useful? I know you used previous code which does not have summary. As you worked on it, if you can write summary, that will be great

Reply:
done